Assets/Mirror/Runtime/SyncList.cs

```diff
@@ -314,9 +324,11 @@ public void Insert(int index, T item)
 
         public bool Remove(T item)
         {
-            bool result = objects.Remove(item);
+            int index = IndexOf(item);
+            bool result = index >= 0;
             if (result)
             {
+                objects.RemoveAt(index);
                 AddOperation(Operation.OP_REMOVE, 0, item);
```

Comment:
This would be more efficient if you simply call ```this.RemoveAt(index)```
you can get rid of OP_REMOVE altogether